### PR TITLE
feat: add manual layer for v1 endpoint

### PR DESCRIPTION
**Pull request by shollyman**
* ~~adds a per-endpoint system session in nox~~
* migrate the manual client/reader work for v1
* TableReferences went away as a first class message, now just a
  formatted string
* changes to read rows
  * estimated_row_count removed due to differences in reported status
  * you no longer have to deal with a StreamPosition message, instead
    there's just a stream name and an offset as top-level request fields
* session creation changes
  * you now provide a prototypical ReadSession message when requesting
    a new read session, and most options (like selected fields, table
    modifiers, and data format) have moved inside it.
  * requested_streams -> max_stream_count

There's additional changes to the surface, but there wasn't much manual
help in front of it.

### CODE REVIEW

**Comment by shollyman** [comment]
failure related to coverage. will poke at that next.

File: google/cloud/bigquery_storage_v1/reader.py [hunk 1]
```diff
@@ -0,0 +1,608 @@
+# Copyright 2020 Google LLC
```
Comment (tswast) [inline google/cloud/bigquery_storage_v1/reader.py]: Are we doing anything to keep this in sync with v1beta1? Fine if no, but if yes let's add a DO NOT EDIT comment near the top.

File: google/cloud/bigquery_storage_v1/reader.py [hunk 2]
```diff
@@ -0,0 +1,608 @@
...
+    def _reconnect(self):
+        """Reconnect to the ReadRows stream using the most recent offset."""
+        self._wrapped = self._client.read_rows(
+            self._name, self._offset, **self._read_rows_kwargs
+        )
+
+    def rows(self, read_session):
+        """Iterate over all rows in the stream.
+
+        This method requires the fastavro library in order to parse row
```
Comment (tswast) [inline google/cloud/bigquery_storage_v1/reader.py]: Looks like I never updated this comment to account for arrow streams.

File: google/cloud/bigquery_storage_v1/reader.py (hunk 1 above)
Comment (shollyman) [reply google/cloud/bigquery_storage_v1/reader.py]: I'd consider keeping them sync if there weren't significant changes in the requests/responses, but with the differences I think it's easier to maintain them independently.

File: google/cloud/bigquery_storage_v1/reader.py (hunk 2 above)
Comment (shollyman) [reply google/cloud/bigquery_storage_v1/reader.py]: updated.

File: noxfile.py [hunk 3]
```diff
@@ -92,10 +92,44 @@ def unit(session):
 
 
 @nox.session(python=["2.7", "3.7"])
-def system(session):
-    """Run the system test suite."""
-    system_test_path = os.path.join("tests", "system.py")
-    system_test_folder_path = os.path.join("tests", "system")
+def system_v1beta1(session):
```
Comment (busunkim96) [inline noxfile.py]: Is there a specific reason for splitting up the system tests into two sessions? 

If this needs to be made permanent `noxfile.py` should be added to the list of excludes in `synth.py`.

Comment (shollyman) [reply noxfile.py]: I mostly did it to speed up the testing cycle, as it takes a couple minutes to get through system tests for a single endpoint.  I can drop it now that I'm done, but is this common enough we should consider parameterizing or plumbing some form of nox flag?

Comment (busunkim96) [reply noxfile.py]: Ah I see. Would you mind opening an issue on https://github.com/googleapis/synthtool/issues to discuss additional parametrizing in the noxfile?

Comment (shollyman) [reply noxfile.py]: Filed https://github.com/googleapis/synthtool/issues/427

File: tests/system/v1/conftest.py [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+@pytest.fixture()
+def small_table_reference():
+    return _TABLE_FORMAT.format(
+        "bigquery-public-data", "utility_us", "country_code_iso"
+    )
+
+
+@pytest.fixture(scope="session")
+def local_shakespeare_table_reference(project_id):
+    return _TABLE_FORMAT.format(project_id, "public_samples_copy", "shakespeare")
```
Comment (tswast) [inline tests/system/v1/conftest.py:63]: @shollyman I'm curious why the local copy of `bigquery-public-data.samples.shakespeare`? I'm adding a script to create this table in https://github.com/googleapis/python-bigquery-storage/pull/74

**Pull request merged**